Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -770,3 +806,212 @@ private extension DatabaseSession {
+            }
+        }
+    }
+    
+    func loadReactions(cid: ChannelId, messageId: MessageId, pagination: Pagination) async throws -> [ChatMessageReaction] {
+        try await withCheckedThrowingContinuation { continuation in
+            loadReactions(cid: cid, messageId: messageId, pagination: pagination) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    @discardableResult func loadReplies(cid: ChannelId, messageId: MessageId, pagination: MessagesPagination, paginationStateHandler: MessagesPaginationStateHandling) async throws -> MessageRepliesPayload {
+        try await withCheckedThrowingContinuation { continuation in
+            loadReplies(cid: cid, messageId: messageId, pagination: pagination, paginationStateHandler: paginationStateHandler) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    func pinMessage(messageId: MessageId, pinning: MessagePinning) async throws -> ChatMessage {
+        try await withCheckedThrowingContinuation { continuation in
+            pinMessage(messageId: messageId, pinning: pinning) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    func resendAttachment(with id: AttachmentId) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            restartFailedAttachmentUploading(with: id) { error in
+                continuation.resume(with: error)
+            }
+        }
+    }
+    
+    func resendMessage(with messageId: MessageId) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            resendMessage(with: messageId) { error in
+                continuation.resume(with: error)
+            }
+        }
+    }
+    
+    func search(query: MessageSearchQuery, policy: UpdatePolicy) async throws -> MessageSearchResults {
+        try await withCheckedThrowingContinuation { continuation in
+            search(query: query, policy: policy) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    func translate(messageId: MessageId, to language: TranslationLanguage) async throws -> ChatMessage {
+        try await withCheckedThrowingContinuation { continuation in
+            translate(messageId: messageId, to: language) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    func unpinMessage(messageId: MessageId) async throws -> ChatMessage {
+        try await withCheckedThrowingContinuation { continuation in
+            unpinMessage(messageId: messageId) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    // MARK: -
+    
+    func loadReplies(for parentMessageId: MessageId, pagination: MessagesPagination, cid: ChannelId, paginationStateHandler: MessagesPaginationStateHandling) async throws -> [ChatMessage] {
```

Comment:
Same question here - probably the answer would be the same as the channel updater one.

Reply:
No other reason than keeping the `Chat.swift` has clean as possible